subprojects/docs/src/docs/userguide/running-builds/gradle_daemon.adoc

```diff
@@ -232,31 +232,51 @@ This enables watching the file system for all builds, unless explicitly disabled
 === Troubleshooting file system watching
 
 Limitations::
-We are working on removing the following limitations to make file system watching production ready.
+File system watching currently has the following limitations:
 - If you have symlinks in your build, you won’t get the performance benefits for those locations (we plan to change this in the future).
 - When multiple daemons are running, the idle ones can pick up the changes produced by the others and create large log files with lots of debug info about the changes.
 - On Windows, we don’t support SUBST and network drives (they might work, but we don’t test them yet).
 
-Gradle does not pick up some of my changes.::
+Enable verbose logging::
+You can instruct Gradle to some more information about the state of the virtual file system, and the events received from the file system using the `org.gradle.vfs.verbose` flag.
+This produces the following output at the start and end of the build:
++
+----
+$ gradle assemble --watch-fs -Dorg.gradle.vfs.verbose=true
+Received 3 file system events since last build while watching 1 hierarchies
+Virtual file system retained information about 2 files, 2 directories and 0 missing files since last build
+> Task :compileJava NO-SOURCE
+> Task :processResources NO-SOURCE
+> Task :classes UP-TO-DATE
+> Task :jar UP-TO-DATE
+> Task :assemble UP-TO-DATE
+
+BUILD SUCCESSFUL in 58ms
+1 actionable task: 1 up-to-date
+Received 5 file system events during the current build while watching 1 hierarchies
+Virtual file system retains information about 3 files, 2 directories and 2 missing files until next build
+----
```

Comment:
Ideally this should come from actually executing a build, but I don't see how we could capture reproducible numbers here.